src/angular-app/languageforge/lexicon/settings/configuration/field-unified-view.model.ts

```diff
@@ -94,19 +93,50 @@ export class ConfigurationFieldUnifiedViewModel {
     }
   }
 
+  static ensureRequiredFields(setting: SettingsBase): boolean {
+    return RequiredFields.requiredFieldList().includes(setting.fieldName);
+  }
+
   static selectAllRow(setting: SettingsBase, settings: SettingsBase[], selectAll: SettingsBase): void {
     const roles = RoleType.roles();
+    const requiredFieldList = RequiredFields.requiredFieldList();
     for (const role of roles) {
       setting[role] = setting.isAllRowSelected;
       ConfigurationFieldUnifiedViewModel.checkIfAllRoleColumnSelected(settings, selectAll, role);
     }
     for (const group of setting.groups) {
-      group.show = setting.isAllRowSelected;
+      if (requiredFieldList.includes(setting.fieldName)) {
+        group.show = true;
+      } else {
+        group.show = setting.isAllRowSelected;
+      }
       ConfigurationFieldUnifiedViewModel
         .checkIfAllGroupColumnSelected(settings, selectAll, setting.groups.indexOf(group));
     }
   }
 
+  static checkIfAllEntryFieldsRowSelected(setting: SettingsBase): void {
+    const roles = RoleType.roles();
+    const requiredFieldList = RequiredFields.requiredFieldList();
+    setting.isAllRowSelected = true;
+    for (const role of roles) {
+      if (!setting[role]) {
+        setting.isAllRowSelected = false;
+        break;
+      }
+    }
+    if (setting.isAllRowSelected) {
```

Comment:
Found a bug here. I can get the "Select All" column to disable the lexeme field by the following steps:

1. Uncheck the checkbox for manager (could do this for any role)
2. Uncheck the "Select All" column for a user (a "group" in our code's current terminology)

Result: this `if` statement is never executed, and so the "Select All" logic ends up turning off the checkbox that we're trying to force to be always on.

I'll fix this bug, and while I'm at it I'll add the "can't unselect the lexeme field" logic to roles as well since Chris has indicated that that was indeed the intent of the feature. Which will involve a bit of a code rewrite, so I'm going to set my review status to "changes requested" in order to block this PR from merging until I've finished fixing the bug.

Reply:
Fixed now.